public/js/charts/worstPMGraph.js

```diff
@@ -73,19 +74,26 @@ function readingAverage(dataValues) {
   return averageAll;
 }
 
+// helper function
+function checkIsNAN(number) {
+  return Number.isNaN(number) ? 0 : number;
+}
 // helper function
 const getAverage = (AQReading, type) => {
-  const total = AQReading.map((result) => result[type]);
+  const total = AQReading.map((result) =>
+    Number.isNaN(result) ? 0 : result[type]
```

Comment:
`result` is not supposed to be a _number_, it's an **_object_**. `result[type]` can be a number. Something like:
```js
  const total = AQReading.map((result) => result[type] || 0);
```
Should be enough, no?

Reply:
@kilemensi Yap